hydrator-transforms-lib/src/main/java/co/cask/hydrator/transforms/PythonEvaluator.java

```diff
@@ -0,0 +1,350 @@
+      case RECORD:
+        return encodeRecord((StructuredRecord) object, schema);
+      case UNION:
+        return encodeUnion(object, schema.getUnionSchemas());
+    }
+
+    throw new RuntimeException("Unable decode object with schema " + schema);
+  }
+
+  private Object encodeRecord(StructuredRecord record, Schema schema) {
+    Map<String, Object> map = new HashMap<>();
+    for (Schema.Field field : schema.getFields()) {
+      map.put(field.getName(), encode(record.get(field.getName()), field.getSchema()));
+    }
+    return map;
+  }
+
+  private Object encodeUnion(Object object, List<Schema> unionSchemas) {
+    for (Schema schema : unionSchemas) {
+      try {
+        return encode(object, schema);
+      } catch (Exception e) {
+        // could be ok, just move on and try the next schema
+      }
+    }
+
+    throw new RuntimeException("Unable decode union with schema " + unionSchemas);
+  }
+
+  private Object encodeMap(Map<Object, Object> map, Schema keySchema, Schema valSchema) {
+    Map<Object, Object> encoded = new HashMap<>();
+    for (Map.Entry<Object, Object> entry : map.entrySet()) {
+      encoded.put(encode(entry.getKey(), keySchema), encode(entry.getValue(), valSchema));
+    }
+    return encoded;
+  }
+
+  private Object encodeArray(List list, Schema componentSchema) {
+    List<Object> encoded = new ArrayList<>();
+    for (Object object : list) {
+      encoded.add(encode(object, componentSchema));
+    }
+    return encoded;
+  }
+
+  private Object decode(Object object, Schema schema) {
+    Schema.Type type = schema.getType();
+
+    switch (type) {
+      case NULL:
+      case BOOLEAN:
+      case INT:
+      case LONG:
+      case FLOAT:
+      case DOUBLE:
+      case BYTES:
+      case STRING:
+        return decodeSimpleType(object, schema);
+      case ENUM:
+        break;
+      case ARRAY:
+        return decodeArray((List) object, schema.getComponentSchema());
+      case MAP:
+        Schema keySchema = schema.getMapSchema().getKey();
+        Schema valSchema = schema.getMapSchema().getValue();
+        // Should be fine to cast since schema tells us what it is.
+        //noinspection unchecked
+        return decodeMap((Map<Object, Object>) object, keySchema, valSchema);
+      case RECORD:
+        return decodeRecord((Map) object, schema);
+      case UNION:
+        return decodeUnion(object, schema.getUnionSchemas());
+    }
+
+    throw new RuntimeException("Unable decode object with schema " + schema);
+  }
+
+  private StructuredRecord decodeRecord(Map nativeObject, Schema schema) {
+    StructuredRecord.Builder builder = StructuredRecord.builder(schema);
+    for (Schema.Field field : schema.getFields()) {
+      String fieldName = field.getName();
+      Object fieldVal = nativeObject.get(fieldName);
+      builder.set(fieldName, decode(fieldVal, field.getSchema()));
+    }
+    return builder.build();
+  }
+
+  @SuppressWarnings("RedundantCast")
```

Comment:
I see this comment in JavascriptTransform:
`// cast so that if it's not really a double it will fail. This is possible for unions,
        // where we don't know what the actual type of the object should be.`